### PR TITLE
bump Zephyr to 3.0.0

### PR DESCRIPTION
**Pull request by mniestroj**
Bump Zephyr version in west manifest to just released 3.0.0 version.

TODO:
- [x] Test each platform with each sample, by following README file and verifying the expected behavior (according to README) with real behavior. After positive verification, a checkmark next to the platform/sample can be added. In case of failure, please add a comment to this PR with description about what went wrong or alternatively try to debug this issue.
  - [x] test "nRF91 Development Kit" with:
    - [x] [dfu](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/dfu) sample
    - [x] [hello](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/hello) sample
    - [x] [hello_sporadic](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/hello_sporadic) sample
    - [x] [lightdb get](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/get) sample
    - [x] [lightdb observe](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/observe) sample
    - [x] [lightdb set](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/set) sample
    - [x] [lightdb led](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb_led) sample
    - [x] [lightdb stream](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb_stream) sample
    - [x] [logging](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/logging) sample
    - [x] [settings](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/settings) sample
  - [x] test "ESP32-WROOM-32D" with:
    - [x] [dfu](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/dfu) sample
    - [x] [hello](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/hello) sample
    - [x] [hello_sporadic](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/hello_sporadic) sample
    - [x] [lightdb get](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/get) sample
    - [x] [lightdb observe](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/observe) sample
    - [x] [lightdb set](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/set) sample
    - [x] [lightdb led](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb_led) sample
    - [x] [lightdb stream](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb_stream) sample
    - [x] [logging](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/logging) sample
    - [x] [settings](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/settings) sample
  - [x] test "nRF52840 DK + ESP32-WROOM-32D" with:
    - [x] [dfu](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/dfu) sample
    - [x] [hello](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/hello) sample
    - [x] [hello_sporadic](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/hello_sporadic) sample
    - [x] [lightdb get](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/get) sample
    - [x] [lightdb observe](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/observe) sample
    - [x] [lightdb set](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/set) sample
    - [x] [lightdb led](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb_led) sample
    - [x] [lightdb stream](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb_stream) sample
    - [x] [logging](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/logging) sample
    - [x] [settings](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/settings) sample
  - [x] test "QEMU x86" with:
    - [x] [dfu](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/dfu) sample
    - [x] [hello](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/hello) sample
    - [x] [hello_sporadic](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/hello_sporadic) sample
    - [x] [lightdb get](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/get) sample
    - [x] [lightdb observe](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/observe) sample
    - [x] [lightdb set](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb/set) sample
    - [x] [lightdb led](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb_led) sample
    - [x] [lightdb stream](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/lightdb_stream) sample
    - [x] [logging](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/logging) sample
    - [ ] ~[settings](https://github.com/golioth/zephyr-sdk/tree/bump-zephyr-3.0.0/samples/settings) sample~ Not applicable
- [x] Update device matrix (just before merge to preserve commit ID)

<a href="https://gitpod.io/#https://github.com/golioth/zephyr-sdk/pull/172"></a>

### CODE REVIEW

**Comment by beriberikix** [comment]
How can we check out the branch to test? Is it simply a matter of creating a new project?

```
west init -m https://github.com/golioth/zephyr-sdk.git --mr bump-zephyr-3.0.0 ~/zephyrproject
```

**Comment by mniestroj** [comment]
@beriberikix Exactly!

**Comment by beriberikix** [comment]
I'm currently testing the ESP32-C3:
* Skipped DFU since MCUboot is not supported on this platform
* lightdb_led doesn't seem to work, I received `[00:00:17.477,687] <wrn> golioth_lightdb: Did not enter CBOR map correctly`. Could be related to the CBOR library change?
  * Also LED example doesn't work, I get warning like `<wrn> golioth_lightdb: There is no LED 1 (total 0)` & ` <wrn> golioth_lightdb: Failed to finish decoding: 33 (QCBOR_ERR_NO_MORE_ITEMS)`
* Settings failed, I don't think it's compatible? Ex. error: `<err> net_if: There is no network interface to work with!`

**Comment by szczys** [comment]
I'm testing thingy91 and noticed some issues with the hello_sporadic sample. I'm going to mark that sample as "passed" in this PR, but that is assuming that #175 gets merged in.

**Comment by szczys** [comment]
Testing nRF9160 feather I see the DFU sample looping and redownloading the binary instead of installing it. @ChrisGammell are you able to test an see if you experience the same behavior?

```
[00:04:27.763,763] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 274432
[00:04:28.492,797] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 275456
[00:04:29.144,989] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 276480
[00:04:29.821,838] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 277504
[00:04:30.393,859] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 278528
[00:04:31.010,772] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 279552
[00:04:31.612,884] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 280576
[00:04:32.197,937] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 281600
[00:04:32.797,943] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 282624
[04:33.491,943] <dbg> g[00:04:34.033,935] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 284672
[00:04:34.418,029] <dbg> golioth_dfu.data_received: Received 96 bytes at offset 285696 (last)
[00:04:34.508,392] <inf> golioth_dfu: Requesting upgrade
[00:04:34.508,728] <inf> golioth_dfu: Rebooting in 1 second(s)
ecolioth_dfu.data_received: Received 1024 bytes at offset 283648
uart:~$ ���U��������W_]շ��}�׵}�U���UUUu��U�w�UU�]�7�������}}������]���uW���U����}]W����}}W����U�U���UUwU���W�_]UU��U���U�w�_�����W������_]���W�_]UU���_��U��������_��U��յW��U

uart:~$ *** Booting Zephyr OS build v2.6.99-ncs1-1  ***
[00:00:00.214,965] <inf> golioth_system: Initializing
[00:01:49.716,217] <dbg> golioth_dfu.main: Start DFU sample
[00:01:49.716,369] <inf> golioth_system: Starting connect
[00:01:50.951,843] <inf> golioth_system: Client connected!
[00:01:51.285,980] <dbg> golioth_dfu: Desired
                                      a3 01 1a 62 2b aa 7f 02  78 40 30 34 66 37 32 33 |...b+... x@04f723
                                      61 66 61 35 33 31 31 61  36 64 30 31 36 34 30 62 |afa5311a 6d01640b
                                      66 65 66 32 63 37 32 38  37 30 30 36 38 64 31 37 |fef2c728 70068d17
                                      61 65 65 61 36 33 65 37  36 33 36 37 64 36 39 64 |aeea63e7 6367d69d
                                      65 61 39 34 31 32 30 66  31 30 03 81 a6 01 64 6d |ea94120f 10....dm
                                      61 69 6e 02 65 31 2e 32  2e 33 03 78 40 35 64 66 |ain.e1.2 .3.x@5df
                                      31 32 38 35 38 38 62 63  38 32 63 63 39 32 66 30 |128588bc 82cc92f0
                                      36 32 34 61 38 31 33 33  33 38 34 32 63 31 63 37 |624a8133 3842c1c7
                                      38 39 61 61 61 35 30 35  36 34 37 64 64 31 31 61 |89aaa505 647dd11a
                                      36 61 63 33 62 39 66 30  39 32 66 33 38 04 1a 00 |6ac3b9f0 92f38...
                                      04 59 c8 05 70 2f 2e 75  2f 63 2f 6d 61 69 6e 40 |.Y..p/.u /c/main@
                                      31 2e 32 2e 33 06 67 6d  63 75 62 6f 6f 74       |1.2.3.gm cuboot  
[00:01:51.286,285] <inf> golioth: Manifest sequence-number: 1647028863
[00:01:51.822,540] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 0
[00:01:51.822,692] <inf> mcuboot_util: Swap type: test
[00:01:51.822,692] <inf> golioth_dfu: swap type: test
[00:01:52.450,714] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 1024
[00:01:52.942,657] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 2048
[00:01:53.352,630] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 3072
[00:01:53.719,726] <dbg> golioth_dfu.data_received: Received 1024 bytes at offset 4096```

**Comment by ncmiller** [comment]
> @ncmiller I have force pushed this branch with recent changes. If everything is fine, then I will replace `HEAD` with commit ID of `bump Zephyr to 3.0.0` commit. Or do you want to put a `v0.1.0` tag there?

I think using the `bump Zephyr to 3.0.0` commit is fine in this PR. I will replace it with `v0.1.0` in the PR that creates the release. And we'll just have to make sure not to change any code between the time when this merges and the release PR.

File: west.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 manifest:
   projects:
     - name: zephyr
-      revision: d51a67b7c098c8bb5479a7b87f05dc40fb76e065
+      revision: 4f8d78ceeb436e82f528511998515f6fc137c6cd
```
Comment (ncmiller) [inline west.yml:4]: Can we put `v3.0.0` here instead?

Comment (mniestroj) [reply west.yml:4]: Yes, we can. Checked NCS to see whether they use tags there. They use git tags there, so no reason we should do it differently. Will update this PR.

Comment (mniestroj) [reply west.yml]: Would be good to include a CI check that runs on each root README.rst update, with something like `git merge-base --is-ancestor <commit-ref-from-device-matrix> <ci-commit-ref>` just to check we didn't merge with wrong commit id / tag (this happened already when using GitHub Merge button, as GitHub always does rebase even if fast-forward would work).

**Review by ncmiller — APPROVED** [review]
I'm preapproving this, assuming you will finish out the final `settings` sample tests.

**Pull request merged**